MoneyDataType/AmountConverter.cs

```diff
@@ -31,9 +39,29 @@ public override Amount Read(ref Utf8JsonReader reader, Type typeToConvert, JsonS
                     case CurrencyName:
                         currency = Currency.FromISOCode(reader.GetString());
```

Comment:
If you have access to this from this class, that means you don't need the `IsKnownCurrency` flag on the interface.

Reply:
Not sure what you mean by this statement.  IsKnownCurrency on the interface allows the use of predefined or user Currencies.  It has nothing to do with this method.  When reading from serialised data, the property names determine if this is a user defined or predefined currency.

Because the amount currency is of type `ICurrency` then it must be included on the interface.

Reply:
What I mean is that you know what currencies are known and don't need to add a flag on the interface for it. The information is available without it.

Reply:
You could say the same about colors, but that too has a user flag.  This flag is a short circuit that ensures we didn't need to continually do a lookup for a specific currency.  Regardless, I moved the flag to a method on the provider instead as it's the thing that can do the lookup.
I kind of like this solution better as it gives implementors of the provider a way to check if it's one of their currencies too.